cmd/put-main.go

```diff
@@ -0,0 +1,149 @@
+// Copyright (c) 2015-2024 MinIO, Inc.
+//
+// This file is part of MinIO Object Storage stack
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package cmd
+
+import (
+	"context"
+	"os"
+	"strconv"
+
+	"github.com/dustin/go-humanize"
+	"github.com/minio/cli"
+	"github.com/minio/mc/pkg/probe"
+)
+
+// put command flags.
+var (
+	putFlags = []cli.Flag{
+		cli.IntFlag{
+			Name:  "parallel, P",
+			Usage: "upload number of parts in parallel",
+		},
+		cli.StringFlag{
+			Name:  "part-size, s",
+			Usage: "each part size",
+		},
+	}
+)
+
+// Put command.
+var putCmd = cli.Command{
+	Name:         "put",
+	Usage:        "upload local objects to object storage",
+	Action:       mainPut,
+	OnUsageError: onUsageError,
+	Before:       setGlobalsFromContext,
+	Flags:        append(append(ioFlags, globalFlags...), putFlags...),
+	CustomHelpTemplate: `NAME:
+  {{.HelpName}} - {{.Usage}}
+
+USAGE:
+  {{.HelpName}} [FLAGS] SOURCE [SOURCE...] TARGET
+
+FLAGS:
+  {{range .VisibleFlags}}{{.}}
+  {{end}}
+ENVIRONMENT VARIABLES:
+  MC_ENCRYPT:      list of comma delimited prefixes
+  MC_ENCRYPT_KEY:  list of comma delimited prefix=secret values
+
+EXAMPLES:
+  01. Put a list of objects from local file system to Amazon S3 cloud storage.
```

Comment:
I dont think we can put a list of objects in one shot so we should rephrase the text here. Better help examples could be
```
1. Put an object from local file system to S3 storage
{{.Prompt}} {{.HelpName}} path-to/object ALIAS/BUCKET
2. Put an object from local file system to S3 bucket with name
{{.Prompt}} {{.HelpName}} path-to/object ALIAS/BUCKET/OBJECT-NAME
3. Put an object from local file system to S3 bucket under a prefix
{{.Prompt}} {{.HelpName}} path-to/object ALIAS/BUCKET/PREFIX/
```
just suggestion if looks good.